src/main/java/frc/robot/Robot.java

```diff
@@ -531,6 +570,78 @@ public void turnControlPanel() {
       m_controlPanelSpinAmount -= 1;
   }
 
+  private void processImage() {
+    m_sink.grabFrame(m_sourceMat);
+    if(doVisionProcessing) {
+      if(!m_sourceMat.empty()) {
+        // Crop out the uneccessary parts of the image
+        Mat m_subMat = new Mat(m_sourceMat, scanArea);
+
+        // Converts m_sourceMat to hsv color
+        Imgproc.cvtColor(m_subMat, m_hsvMat, Imgproc.COLOR_BGR2HSV);
+
+        // Filters for only the color of the tape
+        Core.inRange(m_hsvMat, hsvLow, hsvHigh, m_filteredMat);
+
+        // Find contours
+        Imgproc.findContours(m_filteredMat, contours, new Mat(),
+          Imgproc.RETR_TREE, Imgproc.CHAIN_APPROX_SIMPLE);
+
+        if(contours.size() > 0) {
+          MatOfPoint largestContour = contours.get(0);
+          double largestContourSize = Imgproc.contourArea(largestContour);
+          for(MatOfPoint contour : contours) {
+            double size = Imgproc.contourArea(contour);
+            if(size > largestContourSize) {
+              largestContour = contour;
+              largestContourSize = size;
+            }
+          }
+          contours.clear();
```

Comment:
This seems to already be done by the cleanup method.

Reply:
This is used to clear the list so it can be used again to just store the largest contour. It seems odd but there is no method do draw a single contour, thus a list must be used to hold the contour. The list is cleared again in the cleanup for use in the next loop.

Reply:
Oh okay! I feel like I vaguely remember that.